components/resources/UserRating.js

```diff
@@ -1,20 +1,43 @@
-import { useOperation } from "react-openapi-client";
+import { useOperation, useOperationMethod } from "react-openapi-client";
+import { useEffect, useState } from "react";
 import Rating from "@mui/material/Rating";
+import Loader from "../common/Loader";
 
-function UserRating({ resourceId }) {
-  const { loading, data, error } = useOperation(
+function UserRating({ resourceId, onChangeAverage }) {
```

Comment:
This component handles too much stuff. I really got lost trying to using the component in the resource page. I rewrote the component on my PR.